adx/resource_adx_external_table.go

```diff
@@ -0,0 +1,270 @@
+		withParams = append(withParams, fmt.Sprintf("filesPreview=%t", filesPreview.(bool)))
+	}
+	if validateNotEmpty, ok := d.GetOk("validate_not_empty"); ok {
+		withParams = append(withParams, fmt.Sprintf("validateNotEmpty=%t", validateNotEmpty.(bool)))
+	}
+	if dryRun, ok := d.GetOk("dry_run"); ok {
+		withParams = append(withParams, fmt.Sprintf("dryRun=%t", dryRun.(bool)))
+	}
+
+	withClause := ""
+	if len(withParams) > 0 {
+		withClause = fmt.Sprintf("with(%s)", strings.Join(withParams, ", "))
+	}
+
+	createStatement := ""
+	if len(partitions) > 0 && len(pathFormat) > 0 {
+		createStatement = fmt.Sprintf(".create-or-alter external table %s (%s) kind = %s partition by (%s) pathformat = (%s) dataformat = %s ('%s') %s", name, schema, kind, partitions, pathFormat, dataFormat, storageConnectionString, withClause)
+	} else if len(partitions) > 0 && len(pathFormat) == 0 {
+		createStatement = fmt.Sprintf(".create-or-alter external table %s (%s) kind = %s partition by (%s) dataformat = %s ('%s') %s", name, schema, kind, partitions, dataFormat, storageConnectionString, withClause)
+	} else if len(partitions) == 0 && len(pathFormat) > 0 {
+		createStatement = fmt.Sprintf(".create-or-alter external table %s (%s) kind = %s pathformat = (%s) dataformat = %s ('%s') %s", name, schema, kind, pathFormat, dataFormat, storageConnectionString, withClause)
+	} else {
+		createStatement = fmt.Sprintf(".create-or-alter external table %s (%s) kind = %s dataformat = %s ('%s') %s", name, schema, kind, dataFormat, storageConnectionString, withClause)
+	}
+
+	_, err := queryADXMgmt(ctx, meta, clusterConfig, databaseName, createStatement)
+	if err != nil {
+		return diag.Errorf("error creating external table %s (Database %q): %+v", name, databaseName, err)
+	}
+
+	client, err := getADXClient(meta, clusterConfig)
+	if err != nil {
+		return diag.Errorf("error creating adx client connection: %+v", err)
+	}
+	d.SetId(buildADXResourceId(client.Endpoint(), databaseName, "externaltable", name))
+
+	return resourceADXExternalTableRead(ctx, d, meta)
+}
+
+func resourceADXExternalTableRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	clusterConfig := getAndExpandClusterConfigWithDefaults(ctx, d, meta)
+
+	id, err := parseADXExternalTableID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	showCommand := fmt.Sprintf(".show external table %s ", id.Name)
+
+	resultSet, diags := readADXEntity[ADXExternalTable](ctx, meta, clusterConfig, id, showCommand, "external table")
+	if diags.HasError() {
+		return diags
+	}
+
+	if len(resultSet) < 1 {
+		d.SetId("")
+	} else {
+
+		d.Set("name", id.Name)
+		d.Set("database_name", id.DatabaseName)
+		d.Set("storage_connection_string", resultSet[0].ConnectionStrings)
+		d.Set("partitions", resultSet[0].Partitions)
+		d.Set("path_format", resultSet[0].PathFormat)
+		d.Set("folder", resultSet[0].Folder)
+		d.Set("properties", resultSet[0].Properties)
```

Comment:
What about all other attributes? If those are set - wouldn't this generate a perpetual diff?

Reply:
No, there is no perpetual diff. Honestly I can not figure out what we are doing with those read attributes. Other resources like materialized-views and functions also do not read all attributes they have as input parameters.